lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1088,7 +1110,8 @@ public function resetQueryPart(string $queryPartName) : self
      */
     private function getSQLForSelect() : string
     {
-        $query = 'SELECT ' . implode(', ', $this->sqlParts['select']);
+        $query = 'SELECT ' . ($this->sqlParts['distinct'] === true ? 'DISTINCT ' : '') .
```

Comment:
This is already supposed to be a boolean, and if happens not to be, the bug might still go undetected. If you feel unsafe about it being a boolean, then check that and throw if it isn't, or introduce a private accessor to do the type check. If you feel safe, then simply do this:

```suggestion
        $query = 'SELECT ' . ($this->sqlParts['distinct'] ? 'DISTINCT ' : '') .
```